amp.php

```diff
@@ -272,28 +272,53 @@ function amp_correct_query_when_is_front_page( WP_Query $query ) {
 }
 
 /**
- * Whether this is in 'canonical mode.'
+ * Whether this is in 'canonical mode'.
  *
- * Themes can register support for this with `add_theme_support( 'amp' )`.
- * Then, this will change the plugin from 'paired mode,' and it won't use its own templates.
- * Nor output frontend markup like the 'rel' link. If the theme registers support for AMP with:
- * `add_theme_support( 'amp', array( 'template_dir' => 'my-amp-templates' ) )`
- * it will retain 'paired mode.
+ * Themes can register support for this with `add_theme_support( 'amp' )`:
  *
- * @return boolean Whether this is in AMP 'canonical mode'.
+ *      add_theme_support( 'amp' );
+ *
+ * This will serve templates in native AMP by default but the user would be able to change the template mode
+ * from native to paired in the admin. To force only native to be available, such as when you are using AMP components
+ * in your theme templates, do:
+ *
+ *      add_theme_support( 'amp', array(
+ *          'mode' => 'native',
+ *      ) );
+ *
+ * If you want to force AMP to always be served on a given template, you can use the templates_supported arg,
+ * for example to always serve the Category template in AMP:
+ *
+ *      add_theme_support( 'amp', array(
+ *          'templates_supported' => array(
+ *              'is_category' => true,
+ *          ),
+ *      ) );
+ *
+ * Or if you want to force AMP to be used on all templates:
+ *
+ *      add_theme_support( 'amp', array(
+ *          'templates_supported' => 'all',
+ *      ) );
+ *
+ * @return boolean Whether this is in AMP 'canonical' mode, that is whether it is native and there is not separate AMP URL current URL.
  */
 function amp_is_canonical() {
-	$support = get_theme_support( 'amp' );
-	if ( true === $support ) {
-		return true;
+	if ( ! current_theme_supports( 'amp' ) ) {
+		return false;
 	}
+
+	$mode    = 'native';
+	$support = get_theme_support( 'amp' );
```

Comment:
May there be a more differentiating naming for `current_theme_supports` vs. `get_theme_support`? Probably I am missing some of the context but the naming appear a little confusing.

Reply:
The `current_theme_supports()` and `get_theme_support()` are WordPress core functions. They are somewhat confusing. The former returns whether support has been added (boolean), _and_ applies a filter to let plugins control whether support is present. The `get_theme_support()` on the other hand does not apply a filter, but rather it returns the args that were passed in for the original `add_theme_support()` call.